File: desktop-app/internal/app/intel_parser.go
package app

import (
	"fmt"
	"regexp"
	"strings"
	"time"

	"eve-assistant/desktop-app/internal/protocol"
)

var intelSystems = []string{"Jita", "Amarr", "Dodixie", "Rens", "Hek", "Tama", "Niarja", "Uedama", "Perigen Falls", "Aldranette"}
var intelShips = []string{"Rifter", "Merlin", "Drake", "Caracal", "Raven", "Megathron", "Catalyst", "Venture", "Interceptor"}

// ParseIntel intentionally only emits entities from the supplied text; it never fetches or invents EVE data.
func ParseIntel(text, source string, now time.Time) []protocol.IntelFinding {
	text = strings.TrimSpace(text)
	if text == "" {
		return []protocol.IntelFinding{}
	}
	if source == "" {
		source = "clipboard"
	}
	if now.IsZero() {
		now = time.Now().UTC()
	}
	out := make([]protocol.IntelFinding, 0)
	add := func(kind, value string, confidence float64) {
		out = append(out, protocol.IntelFinding{Kind: kind, Value: value, Stance: "unknown", Source: source, Summary: fmt.Sprintf("%s detected", value), ObservedAt: now, ExpiresAt: now.Add(30 * time.Minute), Confidence: confidence})
	}
	for _, value := range intelSystems {
		if regexp.MustCompile(`(?i)\b` + regexp.QuoteMeta(value) + `\b`).MatchString(text) {
			add("system", value, .95)
		}
	}
	for _, value := range intelShips {
		if regexp.MustCompile(`(?i)\b` + regexp.QuoteMeta(value) + `\b`).MatchString(text) {
			add("ship", value, .88)
		}
	}
	lower := strings.ToLower(text)
	stance := ""
	for _, term := range []string{"敌对", "红名", "hostile", "war target", "neut"} {
		if strings.Contains(lower, strings.ToLower(term)) {
			stance = "hostile"
			break
		}
	}
	if stance == "" {
		for _, term := range []string{"友军", "蓝名", "friendly", "allied", "蓝"} {
			if strings.Contains(lower, strings.ToLower(term)) {
				stance = "friendly"
				break
			}
		}
	}
	if stance != "" {
		out = append(out, protocol.IntelFinding{Kind: "stance", Value: stance, Stance: stance, Source: source, Summary: fmt.Sprintf("%s detected (%s)", stance, stance), ObservedAt: now, ExpiresAt: now.Add(30 * time.Minute), Confidence: .82})
	}
	return out
}

File: mobile-app/lib/intel/intel_parser.dart
class IntelFinding {
  final String kind, value, stance, source, summary;
  final DateTime observedAt, expiresAt;
  final double confidence;
  const IntelFinding({required this.kind, required this.value, required this.stance, required this.source, required this.summary, required this.observedAt, required this.expiresAt, required this.confidence});
  Map<String,dynamic> toJson()=>{'kind':kind,'value':value,'stance':stance,'source':source,'observedAt':observedAt.toUtc().toIso8601String(),'expiresAt':expiresAt.toUtc().toIso8601String(),'confidence':confidence,'summary':summary};
}

class IntelParser {
  static const systems = ['Jita','Amarr','Dodixie','Rens','Hek','Tama','Niarja','Uedama','Perigen Falls','Aldranette'];
  static const ships = ['Rifter','Merlin','Drake','Caracal','Raven','Megathron','Catalyst','Venture','Interceptor'];
  static const hostile = ['敌对','红名','hostile','neutral?','war target','neut'];
  static const friendly = ['友军','蓝名','friendly','allied','蓝'];

  List<IntelFinding> parse(String text, {String source='clipboard', DateTime? now, Duration ttl=const Duration(minutes: 30)}) {
    final observed = now ?? DateTime.now(); final out=<IntelFinding>[];
    for (final s in systems) { if (RegExp('\\b${RegExp.escape(s)}\\b', caseSensitive:false).hasMatch(text)) out.add(_f('system',s,text,source,observed,ttl,0.95)); }
    for (final s in ships) { if (RegExp('\\b${RegExp.escape(s)}\\b', caseSensitive:false).hasMatch(text)) out.add(_f('ship',s,text,source,observed,ttl,0.88)); }
    var stance='unknown';
    if (hostile.any((x)=>text.toLowerCase().contains(x.toLowerCase()))) {
      stance='hostile';
    } else if (friendly.any((x)=>text.toLowerCase().contains(x.toLowerCase()))) {
      stance='friendly';
    }
    if (stance!='unknown') out.add(_f('stance',stance,text,source,observed,ttl,0.82));
    return out;
  }
  IntelFinding _f(String kind,String value,String text,String source,DateTime at,Duration ttl,double confidence)=>IntelFinding(kind:kind,value:value,stance:kind=='stance'?value:'unknown',source:source,summary:'$value detected${kind=='stance'?' ($value)':''}',observedAt:at,expiresAt:at.add(ttl),confidence:confidence);
}
